lib/game_rules.rb

```diff
@@ -0,0 +1,30 @@
+require_relative './token'
+require_relative './board_transformer'
+
+module GameRules
+    include Token
+    include BoardTransformer
+
+    def win?(board:) 
+        horizontal_wins = any_row_a_win?(horizontal_rows(board))
+
+        vertical_wins = any_row_a_win?(vertical_rows(board))
+
+        diagonal_wins = any_row_a_win?(diagonal_rows(board))
+
+        return [ horizontal_wins, vertical_wins, diagonal_wins ].any?
+    end
+
+    def tie?(board:)
+        return board.full?
```

Comment:
A full board does not necessarily mean there is a tie...

Reply:
wait - why not? I would check for tie after wins -- so if its not a win, and the board is full -- wouldnt that be a tie?

Reply:
I think that's the point though - we aren't checking for a win inside `tie?` and you're leaving it up to any client of `GameRules` to assume that they need to check for `win?` first.

Reply:
ahh i see, ok , updating!